front/app/hooks/useFeatureFlags.ts

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from 'react';
+import {
+  currentAppConfigurationStream,
+  IAppConfiguration,
+  TAppConfigurationSetting,
+} from 'services/appConfiguration';
+
+type Parameters = {
+  names: TAppConfigurationSetting[];
+  onlyCheckAllowed?: boolean;
+  onlyOneFeatureFlagRequired?: boolean;
+};
+
+// copied and modified from front/app/hooks/useFeatureFlag.ts
+export default function useFeatureFlags({
+  names,
+  onlyCheckAllowed,
+  onlyOneFeatureFlagRequired = true,
```

Comment:
Why give the default value if it should be consciously set?
I suppose the default value should be the same as the most popular use case. Currently, it can be only `true`.

Reply:
Why not incorporate it directly in your code then, if it's the most popular use case? 🙂 

The edge case here is to require all feature flags. That would be the option (the not often chosen one), so that would default to false.

Another argument for doing it that way is that we most often in the code base have the default be false of such optional props. You can search for `= false,` in the front folder to find them. In the places where we use the opposite, `= true,` we use it imo in the wrong way. 

You could then have something like this instead:

```
export default function useFeatureFlags({
  names,
  onlyCheckAllowed = false,
  allFeatureFlagsRequired = false,
}: Parameters) {
  ... 
  
  return (
    names.length === 0 ||
    (allFeatureFlagsRequired
      ? names.every(isFeatureActive)
      : names.some(isFeatureActive))
  );
}
```

But in fact, I realized we currently don't use the prop, so why have it at all?

Reply:
If it's the most popular use case, why not make it part of the standard behaviour then and leave the edge cases for config option (that would default to false, since they're the edge cases).

But in fact, I realized we currently don't use the prop, so why have it at all? If it's true by default, optional and not used, the prop is not needed at the moment, right? 🙂

Reply:
> why not make it part of the standard behaviour then and leave the edge cases for a config option/prop

I do exactly that :slightly_smiling_face:

> that would default to false, since they're the edge cases

But I do not get this logic. Why should default value be `false` and not `true`?

> why have it at all?

You proposed to use it :slightly_smiling_face: Also, without it and with the name `useFeatureFlags` it's not clear whether we check one or all features. 

It seems we discuss this tiny piece of code too much and now I'm a bit confused about how you want to see the result :smile: 
So, feel free to apply any change you want. But please don't forget to add the ticket name to the commit and reset with `--hard` to the origin branch to avoid messing git history because I pushed with `--force` to this branch.

Reply:
The standard behaviour is to not use all feature flag (use .some on the array), so you don't need onlyOneFeatureFlagRequired. It's not used. I thought it was perhaps earlier?

So you don't need this prop now, just check with `some`. Not sure we'll ever need `every`. You ain't gonna need it. 🙂

We'll stop the discussion about false/true as default value as it isn't needed here.

Reply:
fixed